src/core/ddsc/tests/psmxif.c

```diff
@@ -0,0 +1,441 @@
+/// - Query whether shared memory is supported.
+/// - Assert that there is exactly one psmx instance.
+/// - Assert that the psmx instance has a nonempty instance_name.
+/// - Create some entities
+/// - Check if shared memory is enabled.
+/// - Expectation: Shared memory is enabled iff the psmx interface supports it (use queried value).
+/// - Delete the domain
+/// - Check the function call counts of the dummy psmx.
+/// - Expectation: The counts match expectations. In particular, create counts must match their delete counterpart.
+/// 
+CU_Test(ddsc_psmxif, shared_memory)
+{
+  dummy_mockstats_t* dmock = NULL;
+  char strbuf[512];
+  const size_t strbuf_size = sizeof(strbuf);
+  {
+    // Check that the data types I'm planning to use are actually suitable for use with shared memory.
+    dds_data_type_properties_t props;
+    props = dds_stream_data_types(SC_Model_desc.m_ops);
+    CU_ASSERT_FATAL((props & DDS_DATA_TYPE_IS_MEMCPY_SAFE) == DDS_DATA_TYPE_IS_MEMCPY_SAFE);
+    props = dds_stream_data_types(PsmxType1_desc.m_ops);
+    CU_ASSERT_FATAL((props & DDS_DATA_TYPE_IS_MEMCPY_SAFE) == DDS_DATA_TYPE_IS_MEMCPY_SAFE);
+  }
+
+  const dds_domainid_t domainId = 0;
+  bool supports_shared_memory_expected = true;
+  dds_psmx_endpoint_t* psmx_endpt_expected = NULL;
+  dds_psmx_topic_t* psmx_topic_expected = NULL;
+  bool specify_locator[] = {true, false};
+  uint8_t locator_in[16];
+  memset(locator_in, 0x0, sizeof(locator_in)); // Avoid warning 'uninitialized value'.
+  ((uint64_t*)locator_in)[0] = (uint64_t)0x4a4d203df6996395;
+  ((uint64_t*)locator_in)[1] = (uint64_t)0xe1412fbecc2de4b6;
+
+  int N = sizeof(specify_locator) / sizeof(specify_locator[0]);
+  for (int i = 0; i < N; ++i)
+  {
+    const uint8_t* locator = specify_locator[i] ? locator_in : NULL;
+    char* configstr_in = create_config(domainId, "dummy", locator);
+    const dds_entity_t domain = dds_create_domain(domainId, configstr_in);
+    CU_ASSERT_FATAL(domain > 0);
+    const dds_entity_t participant = dds_create_participant(domainId, NULL, NULL);
+    CU_ASSERT_FATAL(participant > 0);
+    {
+      // Query whether shared memory is supported.
+      dds_domain* dom = NULL;
+      {
+        dds_entity* x = NULL;
+        dds_return_t rc = dds_entity_pin(domain, &x);
+        CU_ASSERT_FATAL(rc == DDS_RETCODE_OK && dds_entity_kind(x) == DDS_KIND_DOMAIN);
+        dom = (dds_domain*)x;
+      }
+      assert(dom->psmx_instances.length >= 1);
+      struct dummy_psmx* dpsmx = (struct dummy_psmx*)dom->psmx_instances.instances[0];
+      dmock = dpsmx->mockstats_get_ptr();
+      CU_ASSERT_FATAL(dmock->cnt_create_psmx == 1); // Confirm the dummy psmx has been loaded.
+      dds_entity_unpin(&dom->m_entity);
+    }
+    // Check that the config string passed to `dds_create_domain()` has been correctly forwarded to the dummy psmx.
+    CU_ASSERT_FATAL(strstr(configstr_in, dmock->config) != NULL); // dmock->config is a substring of the original xml.
```

Comment:
Well, yes, it is a substring. But we know *exactly* what contents it has: those that were passed in. A exact check is just as easy and doesn't require a comment. So that is what it should be.